### DISCUSSION TITLE
Failed to export logs. Server responded with gRPC status code 2. Error message: FRAME_SIZE_ERROR: 4740180

### DISCUSSION THREAD

**Discussion by uliss3s** [open-telemetry/opentelemetry-java#5818] [Q&A]
I'm using the otel java agent (auto instrumentation) in 5 applications running on tomcat. 
The otel collector is running on a docker container. 
If i restart the otel collector container, a specific application stop sending logs, metrics and trace data until i restart tomcat.

I really don't know where to start to try to understand and solve the problem.

otel-java-agent: 1.29.0

log:
```
[otel.javaagent 2023-09-11 12:43:49:596 -0400] [OkHttp http://collector-service:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export logs. Server responded with gRPC status code 2. Error message: FRAME_SIZE_ERROR: 4740180
[otel.javaagent 2023-09-11 12:43:50:705 -0400] [OkHttp http://collector-service:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export spans. Server responded with gRPC status code 2. Error message: FRAME_SIZE_ERROR: 4740180
[otel.javaagent 2023-09-11 12:43:53:404 -0400] [OkHttp http://collector-service:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export metrics. Server responded with gRPC status code 2. Error message: java.io.IOException: FRAME_SIZE_ERROR: 4740180
[otel.javaagent 2023-09-11 12:44:05:717 -0400] [OkHttp http://collector-service:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export spans. Server responded with gRPC status code 2. Error message: Broken pipe
```

**Comment by trask** [comment]
hey @uliss3s! check out prior issues and discussions about this error: https://github.com/search?q=org%3Aopen-telemetry%204740180&type=code

and let us know if that helps resolve the issue for you, thx

**Reply by garydgregory** [reply]
Hi. The URL above is broken. I'm trying to debug a different "2" error...
Actually, I can reproduce this: `Server responded with gRPC status code 2. Error message: FRAME_SIZE_ERROR: 4740180` using prometheus.exe `(version=2.51.2, branch=HEAD, revision=b4c0ab52c3e9b940ab803581ddae9b3d9a452337)` and 
```
      <dependency>
        <groupId>io.opentelemetry</groupId>
        <artifactId>opentelemetry-bom</artifactId>
        <version>1.37.0</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>      
      <dependency>
        <groupId>io.opentelemetry</groupId>
        <artifactId>opentelemetry-bom-alpha</artifactId>
        <version>1.37.0-alpha</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
      <dependency>
        <groupId>io.opentelemetry.instrumentation</groupId>
        <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
        <version>2.3.0-alpha</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>      
```

**Comment by sharmar11** [comment]
I am also getting above FRAME_SIZE_ERROR, is anyone found solution for it?

**Reply by jaydeluca** [reply]
this error usually indicates that you are unable to communicate with the collector via grpc. It can be the result of many different things - for example you are trying to use the wrong protocol (http vs grpc), you aren't pointing at the right address, the collector is misconfigured, or something else impacting connectivity from the app to the collector.
